.dev/roxygen_test.R

```diff
@@ -38,10 +38,8 @@ check_roxygenize_idempotent <- function(LOCALE) {
       next
     }
     cat(sprintf("roxygenize() output differs from saved output for %s.\n", file))
-    cat("Here's the Rd2txt() output for what's saved:\n")
-    Rd2txt(old_file)
-    cat("Here's the Rd2txt() output for roxygenize():\n")
-    Rd2txt(new_file)
+    cat("Here's the tools::Rdiff() comparison of the two files:\n")
+    tools::Rdiff(old_file, new_file)
```

Comment:
``` r
tmp1 <- withr::local_tempfile(lines = c("a", "b"))
tmp2 <- withr::local_tempfile(lines = c("a", "b", "c"))
tools::Rdiff(tmp1, tmp2)
#> files differ in number of lines
```

<sup>Created on 2023-12-19 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

That's not very helpful, either...

Reply:
Ah, that's annoying, thanks for checking. Maybe we should use `testthat::expect_identical()` to handle giving a summarized diff instead?

Reply:
What about
``` r
tmp1 <- withr::local_tempfile(lines = c("a", "b"))
tmp2 <- withr::local_tempfile(lines = c("a", "b", "c"))
system2("diff", c("--unified", normalizePath(tmp1), normalizePath(tmp2)))
```

<sup>Created on 2023-12-19 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
> ```r
> tmp1 <- withr::local_tempfile(lines = c("a", "b"))
> tmp2 <- withr::local_tempfile(lines = c("a", "b", "c"))
> tools::Rdiff(tmp1, tmp2)
> #> files differ in number of lines
> ```
> 
> Created on 2023-12-19 with [reprex v2.0.2](https://reprex.tidyverse.org)
> 
> That's not very helpful, either...

Hmm, this truncated the output:

```
files differ in number of lines:
2a3
> c
```

Note that `tools::Rdiff()` is calling `diff` under the hood, just with different options (`-bw` vs. your `--unified`):

https://github.com/r-devel/r-svn/blob/c786526c27271c4bb04b621d2bde5cdd93033bc0/src/library/tools/R/testing.R#L290-L293

Vs. testthat:

```
testthat::expect_identical(readLines(tmp1), readLines(tmp2))
> Error: readLines(tmp1) (`actual`) not identical to readLines(tmp2) (`expected`).

`actual`:   "a" "b"    
`expected`: "a" "b" "c"
```

(with color if we enable it)

Reply:
Oh, weird. I don't get the output from Rdiff in either my IDE or reprex.
I prefer `--unified` because that's basically `git diff`, which we're very used to reading.

Reply:
What about with `Rdiff(Log = TRUE)` maybe? (just trying to avoid writing our own `system2()` which feels fragile)

Reply:
Let's try it by forcing a failure?

Reply:
Indeed plain `Rdiff()` output is junk:

![image](https://github.com/r-lib/lintr/assets/7606389/54662da5-c3cb-40c1-bf0d-27a3b943eca4)

Trying with Log=TRUE now.

Reply:
OK, `--unified` looks like the way to go. One last small bit is to try and make this part comprehensible:

![image](https://github.com/r-lib/lintr/assets/7606389/64e81c1d-4beb-4fbb-b67c-ab2f926892ba)

Reply:
I wouldn't worry too much about that, unless there is a very simple option for diff to fake this path.

Reply:
more importantly we need to make clear which output comes from which source. handled already at current commit.